src/main/java/net/minecraftforge/common/ForgeHooks.java

```diff
@@ -125,6 +127,30 @@
 
 public class ForgeHooks
 {
+    public static class EntitySelectorProxy
+    {
+        public static <T extends Entity> List<T> matchEntities(ICommandSender sender, String token, Class <? extends T > targetClass)
+        {
+            return EntitySelector.matchEntitiesDefault(sender, token, targetClass);
+        }
+
+        public static boolean matchesMultiplePlayers(String selectorStr)
+        {
+            return EntitySelector.matchesMultiplePlayersDefault(selectorStr);
+        }
+
+        public static boolean hasArguments(String selectorStr)
+        {
+            return EntitySelector.hasArgumentsDefault(selectorStr);
+        }
+    }
+
+    /**
+     * Allows you to override the behavior of EnitySelector by setting this proxy to a custom implementation
+     * Note: You should generally override all three methods to produce consistent results
+     */
+    public static EntitySelectorProxy entitySelectorProxy = new EntitySelectorProxy();
```

Comment:
I suggest to add getter and setter for this. Report when this is set twice by different mods.